src/core/common/message.hpp

```diff
@@ -90,6 +94,7 @@ struct MessageListEntry
 struct BufferHeader
 {
     class Buffer *mNext;  ///< A pointer to the next Message buffer.
+    otContext* mContext;  ///< A pointer to the OpenThread context.
```

Comment:
This design causes every buffer size to be increased by the size of a pointer, which  is not trivial. The other option here was to update every single Message function to take the current context pointer as an arg. Does anyone have a preference as to which design we use?